src/PackageDependencyViewExtension/PackageDependencyView.xaml.cs

```diff
@@ -0,0 +1,232 @@
+        /// </summary>
+        /// <param name="obj">workspace model</param>
+        internal void OnWorkspaceCleared(IWorkspaceModel obj)
+        {
+            if (obj is WorkspaceModel)
+            {
+                // Clear the dependency table.
+                table.Columns.Clear();
+            }
+        }
+
+        private void OnWorkspacePropertyChanged(object sender, PropertyChangedEventArgs args)
+        {
+            if (args.PropertyName == nameof(currentWorkspace.PackageDependencies))
+                DependencyRegen(currentWorkspace);
+        }
+
+        /// <summary>
+        /// Regenerate dependency table
+        /// </summary>
+        /// <param name="ws">workspace model</param>
+        internal void DependencyRegen(WorkspaceModel ws)
+        {
+            // Clear the dependency table.
+            table.Columns.Clear();
+            foreach (var package in ws.PackageDependencies)
+            {
+                bool matchFound = false;
+                // Check if the target package is installed and loaded.
+                foreach (var loadedPackage in ws.LoadedPackageDependencies)
+                {
+                    if (package.Equals(loadedPackage))
+                    {
+                        table.Columns.Add(new Column()
+                        {
+                            ColumnsData = new ObservableCollection<ColumnData>()
+                            {
+                                new ColumnData(package.Name),
+                                new ColumnData(package.Version.ToString(), 100)
+                            }
+                        });
+                        matchFound = true;
+                        continue;
+                    }
+                }
+                // TODO: Not ideal! O(N * M) complexicty, would like LoadedPackageDependencies to be a dictionary or something with constant package name search time
+                if (!matchFound)
+                {
+                    table.Columns.Add(new Column()
+                    {
+                        ColumnsData = new ObservableCollection<ColumnData>()
+                        {
+                            new ColumnData(package.Name, ColumnData.WarningColor),
+                            new ColumnData(package.Version.ToString(), 100)
+                        }
+                    });
+                }
+            }
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="p">ViewLoadedParams</param>
+        public PackageDependencyView(ViewLoadedParams p)
+        {
+            InitializeComponent();
+            DataContext = table;
+            currentWorkspace = p.CurrentWorkspaceModel as WorkspaceModel;
+            p.CurrentWorkspaceChanged += OnWorkspaceChanged;
+            p.CurrentWorkspaceCleared += OnWorkspaceCleared;
+            currentWorkspace.PropertyChanged += OnWorkspacePropertyChanged;
```

Comment:
you need to unsubscribe this event handler.

Reply:
I see how this could always be the same workspace - but I think for safety's sake we should unsub this when the workspace changes and re sub it there.

Reply:
yup, I removed because this is causing crashes sometimes (I think I showed you).. Let me add it back and test